src/framework/handlers/handler.js

```diff
@@ -7,16 +7,42 @@
  */
 
 /**
- * @interface
- * @name ResourceHandler
- * @description Interface for ResourceHandlers used by {@link ResourceLoader}.
+ * Base class for ResourceHandlers used by {@link ResourceLoader}.
  */
 class ResourceHandler {
     /**
-     * @function
-     * @name ResourceHandler#load
-     * @description Load a resource from a remote URL. When loaded (or failed),
-     * use the callback to return an the raw resource data (or error).
+     * Type of the resource the handler handles.
+     *
+     * @type {string}
+     */
+    handlerType;
+
+    /** @private */
+    _maxRetries = 0;
+
+    /**
+     * @param {string} type - The type of resource the handler will load.
+     */
+    constructor(type) {
+        this.handlerType = type;
+    }
+
+    /**
+     * The number of times to retry a failed request for the resource.
+     *
+     * @type {number}
+     */
+    set maxRetries(value) {
```

Comment:
not sure we need the getting and the setter, just a public property should suffice.

Reply:
The getter/setter pair is overridden in some subclasses.